plugins/processors/prometheusadapter/processor.go

```diff
@@ -5,24 +5,195 @@ package prometheusadapter
 
 import (
 	"context"
+	"os"
+	"strings"
 
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.uber.org/zap"
+
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/prometheusadapter/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheus"
+	"github.com/prometheus/common/model"
 )
 
 type prometheusAdapterProcessor struct {
 	*Config
-	logger *zap.Logger
+	logger          *zap.Logger
+	deltaCalculator *internal.DeltaCalculator
 }
 
 func newPrometheusAdapterProcessor(config *Config, logger *zap.Logger) *prometheusAdapterProcessor {
 	d := &prometheusAdapterProcessor{
-		Config: config,
-		logger: logger,
+		Config:          config,
+		logger:          logger,
+		deltaCalculator: internal.NewDeltaCalculator(),
 	}
 	return d
 }
 
 func (d *prometheusAdapterProcessor) processMetrics(_ context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
+
+	d.preprocessFilter(md)
+
+	md.ResourceMetrics().RemoveIf(func(rm pmetric.ResourceMetrics) bool {
+		rma := rm.Resource().Attributes()
+		sms := rm.ScopeMetrics()
+		sms.RemoveIf(func(sm pmetric.ScopeMetrics) bool {
+			metrics := sm.Metrics()
+			metrics.RemoveIf(func(m pmetric.Metric) bool {
+				return d.processMetric(m, rma)
+			})
+			return metrics.Len() == 0
+		})
+		return sms.Len() == 0
+	})
+
+	d.postprocessFilter(md)
+
 	return md, nil
 }
+
+func (d *prometheusAdapterProcessor) preprocessFilter(md pmetric.Metrics) {
```

Comment:
nit: Why doesn't this follow the same `RemoveIf` pattern that the main `processMetrics` does? Seems like it'd be useful to remove the resource metrics if none of the metrics are kept.